src/main/java/com/coreos/jetcd/resolver/DiscoveryEtcdNameResolver.java

```diff
@@ -0,0 +1,89 @@
+package com.coreos.jetcd.resolver;
+
+import com.google.common.base.Preconditions;
+import io.grpc.Attributes;
+import io.grpc.NameResolver;
+import io.grpc.internal.SharedResourceHolder;
+import io.grpc.internal.SharedResourceHolder.Resource;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import java.net.URI;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+
+/**
+ * A discovery service based {@link NameResolver}. DiscoveryEtcdNameResolver will
+ * query configured discovery URL periodically to get endpoints addresses.
+ */
+public class DiscoveryEtcdNameResolver extends NameResolver {
```

Comment:
i think the first naming resolver would be a dns srv resolver with client prefix. see https://github.com/coreos/etcd/blob/master/Documentation/op-guide/clustering.md#dns-discovery